pkg/storegateway/series_refs.go

```diff
@@ -1268,6 +1268,7 @@ func fetchCachedSeriesForPostings(ctx context.Context, userID string, indexCache
 		logSeriesForPostingsCacheEvent(ctx, logger, userID, blockID, shard, itemID, "msg", "can't decode series cache", "err", err)
 		return seriesChunkRefsSet{}, false
 	}
+	defer entry.FreeBuffer()
```

Comment:
I don't think we need this: `entry` is a Protobuf message that's come from a cache, not a gRPC response.

This also means the labels-related changes below are unnecessary.

Reply:
Thanks for noticing, removed it :)